bin/p2-dsctl/main.go

```diff
@@ -0,0 +1,237 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"strconv"
+
+	"gopkg.in/alecthomas/kingpin.v2"
+
+	"github.com/Sirupsen/logrus"
+	ds_fields "github.com/square/p2/pkg/ds/fields"
+	"github.com/square/p2/pkg/kp"
+	"github.com/square/p2/pkg/kp/dsstore"
+	"github.com/square/p2/pkg/kp/flags"
+	"github.com/square/p2/pkg/logging"
+	pc_fields "github.com/square/p2/pkg/pc/fields"
+	"github.com/square/p2/pkg/pods"
+	"github.com/square/p2/pkg/types"
+	"github.com/square/p2/pkg/util"
+	"k8s.io/kubernetes/pkg/labels"
+)
+
+const (
+	CmdCreate  = "create"
+	CmdGet     = "get"
+	CmdList    = "list"
+	CmdEnable  = "enable"
+	CmdDisable = "disable"
+	CmdDelete  = "delete"
+	CmdUpdate  = "update"
+)
+
+var (
+	cmdCreate       = kingpin.Command(CmdCreate, "Create a daemon set.")
+	createAZ        = cmdCreate.Flag("az", "The availability zone of the pod cluster").Required().String()
+	createManifest  = cmdCreate.Flag("manifest", "Complete manifest - must parse as JSON!").Required().String()
+	createMinHealth = cmdCreate.Flag("minhealth", "The minimum health of the daemon set").Required().String()
+	createName      = cmdCreate.Flag("name", "The cluster name (ie. staging, production)").Required().String()
+	createPodID     = cmdCreate.Flag("pod", "The pod ID on the daemon set").Required().String()
+
+	cmdGet = kingpin.Command(CmdGet, "Show a daemon set.")
+	getID  = cmdGet.Arg("id", "The uuid for the daemon set").Required().String()
+
+	cmdList = kingpin.Command(CmdList, "List daemon sets.")
+
+	cmdEnable = kingpin.Command(CmdEnable, "Enable daemon set.")
+	enableID  = cmdEnable.Arg("id", "The uuid for the daemon set").Required().String()
+
+	cmdDisable = kingpin.Command(CmdDisable, "Disable daemon set.")
+	disableID  = cmdDisable.Arg("id", "The uuid for the daemon set").Required().String()
+
+	cmdDelete = kingpin.Command(CmdDelete, "Delete daemon set.")
+	deleteID  = cmdDelete.Arg("id", "The uuid for the daemon set").Required().String()
+
+	cmdUpdate       = kingpin.Command(CmdUpdate, "Update a daemon set.")
+	updateID        = cmdUpdate.Arg("id", "The uuid for the daemon set").Required().String()
+	updateAZ        = cmdUpdate.Flag("az", "The availability zone of the pod cluster").String()
+	updateManifest  = cmdUpdate.Flag("manifest", "Complete manifest - must parse as JSON!").String()
+	updateMinHealth = cmdUpdate.Flag("minhealth", "The minimum health of the daemon set").String()
+	updateName      = cmdUpdate.Flag("name", "The cluster name (ie. staging, production)").String()
+	updatePodID     = cmdUpdate.Flag("pod", "The pod ID on the daemon set").String()
+)
+
+func main() {
+	cmd, consulOpts := flags.ParseWithConsulOptions()
+	client := kp.NewConsulClient(consulOpts)
+	logger := logging.NewLogger(logrus.Fields{})
+	dsstore := dsstore.NewConsul(client, 3, &logger)
+
+	switch cmd {
+	case CmdCreate:
+		az := pc_fields.AvailabilityZone(*createAZ)
+		minHealth, err := strconv.Atoi(*createMinHealth)
```

Comment:
I did this because `updateMinHealth` uses `String()` so it is consistent.
`updateMinHealth` uses `String()` because I want to check `*updateMinHealth==""` so that I would know if the user did not specify the optional argument.